prow/plugins/trigger/generic-comment.go

```diff
@@ -156,13 +157,16 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 						"repo":    repo,
 					})
 					runID := run.ID
-					go func() {
-						if err := c.GitHubClient.TriggerFailedGitHubWorkflow(org, repo, runID); err != nil {
-							log.Errorf("attempt to trigger github run failed: %v", err)
-						} else {
-							log.Infof("successfully triggered action run")
+					if err := c.GitHubClient.TriggerFailedGitHubWorkflow(org, repo, runID); err != nil {
+						log.Errorf("attempt to trigger github run failed: %v", err)
+						if strings.Contains(err.Error(), "Unable to retry this workflow run because it was created over a month ago") {
+							resp := "The workflow triggered has expired. You can find more information about the same here <>"
+							c.Logger.Infof("Commenting \"%s\".", resp)
+							return c.GitHubClient.CreateComment(org, repo, number, plugins.FormatResponseRaw(gc.Body, gc.HTMLURL, gc.User.Login, resp))
```

Comment:
We can't really return here because there might still be runs that can be successfully retried, if we return here, we basically don't do anything the moment we encounter an expired run.

Reply:
can different workflows have different expired timings?

Reply:
@AvineshTripathi good question!
The only scenario that I had in mind while reviewing this was if an admin of the repo manually re triggered a particular workflow alone and not all. In that case, this is a scenario that can happen.